stargz/verify/verify.go

```diff
@@ -120,70 +122,157 @@ func NewVerifiableStagz(sgz *io.SectionReader) (newSgz io.Reader, jtocDigest str
+}
+
 // extractTOCJSON extracts TOC JSON from the given stargz reader, with avoiding
 // scanning the entire blob leveraging its footer. The parsed TOC JSON contains
 // additional fields usable for chunk-level content verification.
-func extractTOCJSON(sgz *io.SectionReader) (toc *jtoc, tocOffset int64, footer io.ReadSeeker, err error) {
+func extractTOCJSON(sgz *io.SectionReader) (toc tocInfo, err error) {
 	// Parse stargz footer and get the offset of TOC JSON
 	if sgz.Size() < stargz.FooterSize {
-		return nil, 0, nil, errors.New("stargz data is too small")
+		return tocInfo{}, errors.New("stargz data is too small")
 	}
 	footerReader := io.NewSectionReader(sgz, sgz.Size()-stargz.FooterSize, stargz.FooterSize)
 	zr, err := gzip.NewReader(footerReader)
 	if err != nil {
-		return nil, 0, nil, errors.Wrap(err, "failed to uncompress footer")
+		return tocInfo{}, errors.Wrap(err, "failed to uncompress footer")
 	}
 	defer zr.Close()
 	if len(zr.Header.Extra) != 22 {
-		return nil, 0, nil, errors.Wrap(err, "invalid extra size; must be 22 bytes")
+		return tocInfo{}, errors.Wrap(err, "invalid extra size; must be 22 bytes")
 	} else if string(zr.Header.Extra[16:]) != "STARGZ" {
-		return nil, 0, nil, errors.New("invalid footer; extra must contain magic string \"STARGZ\"")
+		return tocInfo{}, errors.New("invalid footer; extra must contain magic string \"STARGZ\"")
 	}
-	tocOffset, err = strconv.ParseInt(string(zr.Header.Extra[:16]), 16, 64)
+	tocOffset, err := strconv.ParseInt(string(zr.Header.Extra[:16]), 16, 64)
 	if err != nil {
-		return nil, 0, nil, errors.Wrap(err, "invalid footer; failed to get the offset of TOC JSON")
+		return tocInfo{}, errors.Wrap(err, "invalid footer; failed to get the offset of TOC JSON")
 	} else if tocOffset > sgz.Size() {
-		return nil, 0, nil, fmt.Errorf("invalid footer; offset of TOC JSON is too large (%d > %d)",
+		return tocInfo{}, fmt.Errorf("invalid footer; offset of TOC JSON is too large (%d > %d)",
 			tocOffset, sgz.Size())
 	}
 
 	// Decode the TOC JSON
 	tocReader := io.NewSectionReader(sgz, tocOffset, sgz.Size()-tocOffset-stargz.FooterSize)
 	if err := zr.Reset(tocReader); err != nil {
-		return nil, 0, nil, errors.Wrap(err, "failed to uncompress TOC JSON targz entry")
+		return tocInfo{}, errors.Wrap(err, "failed to uncompress TOC JSON targz entry")
 	}
 	tr := tar.NewReader(zr)
 	h, err := tr.Next()
 	if err != nil {
-		return nil, 0, nil, errors.Wrap(err, "failed to get TOC JSON tar entry")
+		return tocInfo{}, errors.Wrap(err, "failed to get TOC JSON tar entry")
 	} else if h.Name != stargz.TOCTarName {
-		return nil, 0, nil, fmt.Errorf("invalid TOC JSON tar entry name %q; must be %q",
+		return tocInfo{}, fmt.Errorf("invalid TOC JSON tar entry name %q; must be %q",
 			h.Name, stargz.TOCTarName)
 	}
-	toc = new(jtoc)
-	if err := json.NewDecoder(tr).Decode(&toc); err != nil {
-		return nil, 0, nil, errors.Wrap(err, "failed to decode TOC JSON")
+	dgst := sha256.New()
```

Comment:
Use OCI `Digster`

Reply:
Fixed patches to use `Digester` instead of directly using `sha256`, etc.